modules/custom/d_content/modules/d_content_init/src/ContentInitManagerBlock.php

```diff
@@ -31,14 +38,37 @@ class ContentInitManagerBlock extends ContentInitManagerBase {
    *   Logger channel factory.
    * @param \Drupal\Core\Block\BlockManagerInterface $block_manager
    *   Block manager interface.
+   * @param ModuleHandlerInterface $module_handler
+   *   Module handler interface.
    */
   public function __construct(
     EntityTypeManagerInterface $entity_manager,
     SerializationInterface $serialization,
     LoggerChannelFactory $logger_factory,
-    BlockManagerInterface $block_manager) {
+    BlockManagerInterface $block_manager,
+    ModuleHandlerInterface $module_handler) {
     parent::__construct($entity_manager, $serialization, $logger_factory);
     $this->blockManager = $block_manager;
+    $this->moduleHandler = $module_handler;
+  }
+
+  /**
+   * Create block (plugin type).
+   *
+   * @param array $block
+   *   Array with definition of block to create.
+   *
+   * @return array|\Drupal\Core\Entity\EntityInterface
+   *   Created block entity.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  protected function createBlockPlugin(array $block) {
+    $values = $this->getBaseBlockValues($block['info']);
+    $values['id'] = $values['id'] ?? 'block_plugin_' . time();
```

Comment:
hmm, just a thought - what if someone runs this function several time in code giving his block no `'id'` - wouldn't he end up getting several block with the same id - would it even pass without exception? WDYT

Reply:
Yes, I thought about it. 

If there is no ID declaration in the YML file then code will create several blocks with ID based on time(). If time() result will be the same for each function call then ID will be the same and we will get the exception.

We can use UUID instead of time().